provider/dynamodb.go

```diff
@@ -0,0 +1,199 @@
+	log.Infof("Creating DynamoDB client...")
+	sess := session.New()
+	if provider.Region == "" {
+		log.Debug("No Region provided. Using us-east-1...")
+		provider.Region = "us-east-1"
+	}
+	cfg := &aws.Config{
+		Region: &provider.Region,
+		Credentials: credentials.NewChainCredentials(
+			[]credentials.Provider{
+				&credentials.StaticProvider{
+					Value: credentials.Value{
+						AccessKeyID:     provider.AccessKeyID,
+						SecretAccessKey: provider.SecretAccessKey,
+					},
+				},
+				&credentials.EnvProvider{},
+				&credentials.SharedCredentialsProvider{},
+				defaults.RemoteCredProvider(*(defaults.Config()), defaults.Handlers()),
+			}),
+	}
+
+	return &dynamoClient{
+		dynamodb.New(sess, cfg),
+	}
+}
+
+// scanTable scans the given table and returns slice of all items in the table
+func (provider *DynamoDB) scanTable(client *dynamoClient) ([]map[string]*dynamodb.AttributeValue, error) {
+	log.Debugf("Scanning table: %s ...", provider.TableName)
+	params := &dynamodb.ScanInput{
+		TableName: aws.String(provider.TableName),
+	}
+	items := make([]map[string]*dynamodb.AttributeValue, 0)
+	err := client.db.ScanPages(params,
+		func(page *dynamodb.ScanOutput, lastPage bool) bool {
+			items = append(items, page.Items...)
+			return !lastPage
+		})
+	if err != nil {
+		log.Errorf("Failed to scan table %s", provider.TableName)
+		return nil, err
+	}
+	log.Debugf("Successfully scanned %s table", provider.TableName)
+	return items, nil
+}
+
+// loadDynamoConfig retrieves items from dynamodb and converts them into Backends and Frontends in a Configuration
+func (provider *DynamoDB) loadDynamoConfig(client *dynamoClient) (*types.Configuration, error) {
+	items, err := provider.scanTable(client)
+	if err != nil {
+		return nil, err
+	}
+	log.Debugf("Amount of Items: %d", len(items))
+	backends := make(map[string]*types.Backend)
+	frontends := make(map[string]*types.Frontend)
+	// unmarshal dynamoAttributes into Backends and Frontends
+	for i, item := range items {
+		log.Debugf("Item: %d\n%v", i, item)
+		// verify the type of each item by checking to see if it has
+		// the corresponding type, backend or frontend map
+		if backend, exists := item["backend"]; exists {
+			log.Debugf("Unmarshaling backend...")
+			tmpBackend := &types.Backend{}
+			err = dynamodbattribute.Unmarshal(backend, tmpBackend)
+			if err != nil {
+				log.Errorf(err.Error())
+			} else {
+				backends[*item["name"].S] = tmpBackend
+				log.Debugf("Backend unmarshalled successfully")
+			}
+		} else if frontend, exists := item["frontend"]; exists {
+			log.Debugf("Unmarshaling frontend...")
+			tmpFrontend := &types.Frontend{}
+			err = dynamodbattribute.Unmarshal(frontend, tmpFrontend)
+			if err != nil {
+				log.Errorf(err.Error())
+			} else {
+				frontends[*item["name"].S] = tmpFrontend
+				log.Debugf("Frontend unmarshalled successfully")
+			}
+		} else {
+			log.Warnf("Something is up! Following item doesn't follow compatible format:\n%v", item)
```

Comment:
I'd suggest to drop the `\n`: Newlines in error messages often make it hard to do things like grepping for errors easily.